File: Task List/typeScriptFiles/Task.ts
export class Task {
    public id: number;
    public title: string;
    public description: string;
    public completed: boolean;
    public date: Date;

    public getId(): number {
        return this.id;
    }
    public setId(value: number) {
        this.id = value;
    }

    public getTitle(): string {
        return this.title;
    }
    public setTitle(value: string) {
        this.title = value;
    }

    public getDescription(): string {
        return this.description;
    }
    public setDescription(value: string) {
        this.description = value;
    }

    public getCompleted(): boolean {
        return this.completed;
    }
    public setCompleted(value: boolean) {
        this.completed = value;
    }

    public getDate(): Date {
        return this.date;
    }
    public setDate(value: Date | null) {
        this.date = value ? value : new Date();
    }

    constructor(id: number, title: string, description: string, completed: boolean, date: Date) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.completed = completed;
        this.date = date;
    }

    public toString(): string {
        return `ID: ${this.id}, title: ${this.title}, description: ${this.description}, completed: ${this.completed}, date: ${this.date}`;
    }

}

File: Task List/javaFiles/Task.java
package javaFiles;

import java.util.Date;

public class Task {
   private int id;
   private String title;
   private String description;
   private Date date;
   private boolean completed;

    public Task(int id, String title, String description,Date date) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.date = date;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public boolean isCompleted() {
        return completed;
    }

    public void setCompleted(boolean completed) {
        this.completed = completed;
    }

    @Override
    public String toString() {
        return id + " - " + title + " (" + description + ") [" + (completed ? "Completed" : "Pending") + "] " + date;
}

}
